build.py

```diff
@@ -87,6 +87,7 @@ class HelpFormatter(argparse.ArgumentDefaultsHelpFormatter, argparse.RawDescript
     # Default to not building the language bindings
     parser.add_argument("--build_csharp", action="store_true", help="Build the C# API.")
     parser.add_argument("--build_java", action="store_true", help="Build Java bindings.")
+    parser.add_argument("--publish_java_maven", action="store_true", help="Publish Java bindings to local Maven repository after tests.")
```

Comment:
Could we avoid exposing this as an argument to the `build.py` script? Are there good defaults? If this is needed only at the cmake level, shall we ask users to pass it in like `python build.py ... --cmake_extra_defines PUBLISH_JAVA_MAVEN=ON`?

Reply:
It tells the Java build to put it in the user's local Maven repository which will be checked by other Maven & Gradle builds on their system, so it makes it easier for local development. However you don't want it for all the builds, particularly if you're working on a feature or don't plan to use the local build (e.g. in the CI) so defaulting it to be on isn't ideal.